functions/index.js

```diff
@@ -369,7 +370,63 @@ exports.setRoles = functions.https.onCall(async (data, context) => {
   return newUser.customClaims;
 });
 
+
+exports.apiAuth = functions.https.onRequest(async (request, response) => {
+  let key = request.body.key
+  let secret = request.body.secret
+  if (!key || !secret) {
+    return response.status(401).send({error: "Unauthorized"})
+  }
+  const usersRef = db.collection("users");
```

Comment:
Change collection name to "apiUsers" - "users" collection is used for keycaplendar specific settings such as favourites and presets (to be rolled out soon™)